CMake/config.cmake

```diff
@@ -40,29 +40,56 @@ set(README_FILE "${CMAKE_SOURCE_DIR}/README.md")
 set(LICENSE_FILE "${CMAKE_SOURCE_DIR}/LICENSE.txt")
 
 # dirs where the binaries should be placed, installed
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin")
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/lib")
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin")
 set(CMAKE_EXECUTABLE_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/bin")
-set(RTTR_INSTALL_DIR "${CMAKE_BINARY_DIR}/install")
+
+# here we specify the installation directory
+if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+  set(CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/install" CACHE PATH  "RTTR install prefix" FORCE)
+endif()
 
 # in order to group in visual studio the targets into solution filters
-set_property( GLOBAL PROPERTY USE_FOLDERS ON)
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 
 #3rd part dependencies dirs
 set(RTTR_3RD_PARTY_DIR "${CMAKE_SOURCE_DIR}/3rd_party")
 
 getNameOfDir(CMAKE_LIBRARY_OUTPUT_DIRECTORY RTTR_TARGET_BIN_DIR)
 is_vs_based_build(VS_BUILD)
-if (VS_BUILD)
-  #set(RTTR_BIN_INSTALL_DIR ${RTTR_TARGET_BIN_DIR}/\${CMAKE_INSTALL_CONFIG_NAME})
-  set(RTTR_BIN_INSTALL_DIR ${RTTR_TARGET_BIN_DIR})
-  set(RTTR_LIB_INSTALL_DIR "lib")
-else()
-  set(RTTR_BIN_INSTALL_DIR ${RTTR_TARGET_BIN_DIR})
-  set(RTTR_LIB_INSTALL_DIR "lib")
+
+# set all install directories for the targets
+if(UNIX)
+  include(GNUInstallDirs)
+  # all directories set
+else(WINDOWS)
+  set(CMAKE_INSTALL_LIBDIR "lib")
+  set(CMAKE_INSTALL_BINDIR "bin")
+  set(CMAKE_INSTALL_DATADIR "${CMAKE_INSTALL_PREFIX}")
+endif()
+
+set(CMAKE_DEBUG_POSTFIX CACHE STRING "_d")
```

Comment:
for unkown reason the postfix will not be applied to `rttr_core`
![image](https://user-images.githubusercontent.com/11368160/35354149-84c88d00-0149-11e8-8292-1118801222b5.png)

Reply:
The postfix is necessary, otherwise we cannot copy the debug and release libraries, into one folder.

Reply:
for sure! but why is it not working? the CMAKE_DEBUG_POSTFIX is set...?